### PR TITLE
Add darts exercise

### LINKED ISSUE #25: Darts exercise
Claimed.

And https://github.com/exercism/awk/pull/24

### PR DESCRIPTION
**Pull request by IsaacG**
Closes #25

### CODE REVIEW

**Comment by glennj** [comment]
Adding the "Closes" in the description sets the linked issue in the Development section on the right, which auto-closes the issue.

**Comment by IsaacG** [comment]
> Adding the "Closes" in the description sets the linked issue in the Development section on the right, which auto-closes the issue.

Ah. That's handy. Thanks!

File: exercises.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ List extracted from problem-specifications repo on 2022-06-03
 | [counter][counter] | . |
 | [crypto-square][crypto-square] | . |
 | [custom-set][custom-set] | . |
-| [darts][darts] | . |
+| [darts][darts] | [implemented](https://github.com/exercism/awk/pull/24) |
```
Comment (glennj) [inline exercises.md:33]: In hindsight, is this file helpful?

File: exercises/practice/darts/darts.awk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env gawk -f
+
+{
+    distance = sqrt($1 ^ 2 + $2 ^ 2)
+    if (distance <= 1) { print 10 }
+    else if (distance <= 5) { print 5 }
+    else if (distance <= 10) { print 1 }
+    else { print 0 }
+}
```
Comment (glennj) [inline exercises/practice/darts/darts.awk:9]: Stub file?

File: exercises/practice/darts/test-darts.bats [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bats
+load bats-extra
+
+teardown() {
+    rm -f input.txt
+}
+
+@test "Missed target" {
+    #[[ $BATS_RUN_SKIPPED == "true" ]] || skip
+    run gawk -f darts.awk <<< '-9 9'
```
Comment (glennj) [inline exercises/practice/darts/test-darts.bats:10]: The herestring looks nice. Should be more obvious for students.

**Review by glennj — CHANGES_REQUESTED** [review]

File: exercises.md (hunk 1 above)
Comment (IsaacG) [reply exercises.md:33]: Eh. Only as a way to track what exercises are needed, which can be done via Github or the configlet. So ... not super helpful.

File: exercises/practice/darts/darts.awk (hunk 2 above)
Comment (IsaacG) [reply exercises/practice/darts/darts.awk:9]: Woops. Does the validate command overwrite this? Fixing.

Comment (glennj) [reply exercises/practice/darts/darts.awk:6]: It does, then it does a `git checkout` to restore it, so if you haven't committed, it's trashed.

Comment (IsaacG) [reply exercises/practice/darts/darts.awk:6]: Yeah. I ran the validate prior to committing.

**Review by glennj — APPROVED** [review]

**Pull request merged**